### PR TITLE
feat: expand logging

### PR DESCRIPTION
**Pull request by Aleksgav**

### CODE REVIEW

File: object/application.go [hunk 1]
```diff
@@ -372,7 +373,7 @@ func GetMaskedApplications(applications []*Application, userId string) []*Applic
 	return applications
 }
 
-func UpdateApplication(id string, application *Application) (bool, error) {
+func UpdateApplication(ctx *context.Context, id string, application *Application) (bool, error) {
```
Comment (dplynskiy) [inline object/application.go:376]: why context arg with pointer?

File: object/application.go [hunk 2]
```diff
@@ -399,7 +400,10 @@ func UpdateApplication(id string, application *Application) (bool, error) {
 		return false, err
 	}
 
+	record := GetRecord(ctx)
 	for _, providerItem := range application.Providers {
+		record.AddReason(fmt.Sprintf("Added provider: %s", providerItem.Name))
```
Comment (dplynskiy) [inline object/application.go:405]: providers can be same as in oldApplication, so no provider was added, but here we will save it as "added provider"

File: object/application.go (hunk 1 above)
Comment (dplynskiy) [reply object/application.go:376]: another point - this is Beego context, Beego library struct. it's bad dependency for object layer, I think we should avoid this, could you try to find solution with context.Context(go built-in  interface) with record link inside as arg, or with record Object as arg?

File: object/application.go (hunk 2 above)
Comment (dplynskiy) [reply object/application.go:405]: if we really need such record, probably we should find diff between old application providers list and new

File: object/application.go (hunk 1 above)
Comment (Aleksgav) [reply object/application.go]: fixed

File: object/application.go (hunk 2 above)
Comment (Aleksgav) [reply object/application.go]: added diff

**Review by dplynskiy — APPROVED** [review]

**Review by ISulimanov — APPROVED** [review]

**Pull request merged**